backtrace-library/src/main/java/backtraceio/library/breadcrumbs/BacktraceBreadcrumbsLogger.java

```diff
@@ -0,0 +1,104 @@
+package backtraceio.library.breadcrumbs;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+
+import org.json.JSONObject;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+
+import backtraceio.library.enums.BacktraceBreadcrumbLevel;
+import backtraceio.library.enums.BacktraceBreadcrumbType;
+import backtraceio.library.logger.BacktraceLogger;
+
+public class BacktraceBreadcrumbsLogger {
+    /**
+     * The base directory of the breadcrumb logs
+     */
+    final private static String breadcrumbLogDirectory = "breadcrumbs";
+
+    final private static String breadcrumbLogFileName = "bt-breadcrumb-0";
+
+    private final String LOG_TAG = BacktraceBreadcrumbsLogger.class.getSimpleName();
+
+    private long breadcrumbId = 0;
+
+    private BacktraceQueueFileHelper backtraceQueueFileHelper;
+
+    /**
+     * We truncate messages longer than this
+     */
+    private final int maxMessageSizeBytes = 1024;
+
+    /**
+     * We stop adding new attributes once we hit this limit
+     */
+    private final int maxAttributeSizeBytes = 1024;
+
+    public BacktraceBreadcrumbsLogger(Context context, int maxQueueFileSizeBytes) throws IOException, NoSuchMethodException {
+        File breadcrumbLogDir = new File(BacktraceBreadcrumbsLogger.getBreadcrumbLogDirectory(context));
+        breadcrumbLogDir.mkdir();
```

Comment:
I missed it in the first read. We can't use access to files outside Backtrace Database. Its not expected for user sthat we will create/manage files in two different places. Also users who doesn't use database expects no file operations at all.

That being said we should pass database path to breadcrumbs integration and this is where we should store files. It looks like breadcrumbs integration should depends on backtrace database

Reply:
Done, thank you for the advice!